release/ray_release/byod/build_ray.py

```diff
@@ -66,7 +66,7 @@ def _get_build(py_version: str) -> Dict[str, Any]:
     ]
     return {
         "label": py_version,
-        "agents": {"queue": "release_queue_small"},
+        "agents": {"queue": "runner_queue_medium_branch"},
```

Comment:
you also changed it to the "branch" queue rather than the "release" queue here.

Reply:
right, it's on purpose, it's the same queue that was setup to build these dockers so the build should just works

Reply:
which will be using a different role.

Reply:
the branch role also has permission to push docker image so the step will succeed. Or you worry about switching between different aws roles within one pipeline?

Reply:
I am not sure about the security implication, and it is also a bit confusing.

we can:
- either, just kill the `release_queue_small` and use `runner_queue_small_branch` on release tests
- or, add a `release_queue_medium` autoscaling pool.

I am more for killing the `release_queue_small`, as we are already running into autoscaling api quotas right now, because we have too many different pools.

Reply:
Reusing sounds good!

Reply:
could you leave a TODO near a `release_queue_small` somewhere, or file a bug for tracking deprecating the `release_queue_small` pool?

Reply:
TODO added!